pkg/registry/core/next/tail_registry_client.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package next
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/registry"
+	"google.golang.org/grpc"
+)
+
+// tailRegistryClient is a simple implementation of registry.NetworkServiceRegistryClient that is called at the end
+// of a chain to ensure that we never call a method on a nil object
+type tailRegistryClient struct{}
```

Comment:
Do we need the same thing for `next.RegistryServer`?
I guess we need to check that `pkg/registry/next` for `RegistryClient`/`RegistryServer` is working as expected.
Let's cover `pkg/registry/next` by unit tests and fix problems.

Reply:
Yes, we need. I've added it